### PR TITLE
build against and require lru >= 0.3.0

### PR DESCRIPTION
**Pull request by yomimono**
As noted in #27 .

### CODE REVIEW

File: lib/mirage_nat_lru.ml [hunk 1]
```diff
@@ -55,9 +55,7 @@ module Storage = struct
     let lookup t key =
       MProf.Trace.label "Mirage_nat_hashtable.lookup.read";
       let t = L.table t in
-      match L.LRU.find key !t with
-      | None -> Lwt.return_none
-      | Some (v, t') -> t := t'; Lwt.return (Some v)
+      Lwt.return @@ L.LRU.find key !t
```
Comment (pqwy) [inline lib/mirage_nat_lru.ml]: Nice `key` you have there. Would be a shame if it got eagerly pruned.

You sure you don't also want to `promote` it to the most-recently-used?

Comment (yomimono) [reply lib/mirage_nat_lru.ml]: Look, I didn't get into programming for the *reading* and the *understanding* and the *doing the right thing all the time*.

**Comment by hannesm** [comment]
in https://github.com/mirage/mirage-nat/pull/28/files#diff-d05a517e7f6f4ce63b05a5790a81788bR79 you likely want to call `LRU.trim` to get rid of potentially too much data in the store...

File: lib/mirage_nat_lru.ml (hunk 1 above)
Comment (pqwy) [reply lib/mirage_nat_lru.ml]: No, I absolutely understand where you're coming from ;)

**Comment by pqwy** [comment]
One last nit -- one which concerns tcpip too: why is everybody using the functional structure just to stick it in a reference cell (and then wrap in `lwt`)?

Doing that just replicates what the mutable thing [does](https://pqwy.github.io/lru/doc/lru/Lru/index.html#sem). And as it says on the tin, using `Lru.M` is more efficient when applicable.

If the idea was to avoid collision-based complexity attacks, there's a seeded variant of`Lru.M`.

**Comment by pqwy** [comment]
An alternative in #29.

**Comment by hannesm** [comment]
@pqwy 
> One last nit -- one which concerns tcpip too: why is everybody using the functional structure just to stick it in a reference cell (and then wrap in lwt)?

I find functional APIs nicer. The Fragments module in the TCP/IP stack does not keep nor modify state :)
This is pretty nice for [testing](https://github.com/mirage/mirage-tcpip/blob/v3.7.4/test/test_ipv4.ml#L49-L262), although of course this can be done with a mutable LRU as well.

**Comment by pqwy** [comment]
I mean to be sure, a functional representation of *anything* is a winner in my book, as long as you understand the price.

I must have confused tcpip with this. I thought I remembered a `ref` being involved there too.

**Comment by yomimono** [comment]
Having merged #29 I'll close this.

**Pull request closed without merging**